### DISCUSSION TITLE
Replace critters with beasties

### DISCUSSION THREAD

**Discussion by roderickhsiao** [vercel/next.js#72039] [Ideas]
(not a bug, perhaps an enhancement?)
### Link to the code that reproduces this issue

https://github.com/roderickhsiao/with-tailwindcss-app/blob/main/package.json#L12

### To Reproduce

1. enable experiment `optimizeCss` flag
2. Next JS use [critters](https://github.com/vercel/next.js/blob/5a4d159846667595090fd48c263cb37f0155337d/packages/next/src/server/post-process.ts#L38) to optimize CSS
3. critters author marked the package 
```
Ownership of Critters has moved to the Nuxt team, who will be maintaining the project going forward. If you'd like to keep using Critters, please switch to the actively-maintained fork at https://github.com/danielroe/beasties
```

### Current vs. Expected behavior

No warning when opt-in optimize CSS

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.1.0: Thu Oct 10 21:03:11 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T6020
  Available memory (MB): 32768
  Available CPU cores: 12
Binaries:
  Node: 22.11.0
  npm: 10.9.0
  Yarn: 1.22.19
  pnpm: N/A
Relevant Packages:
  next: 15.0.2 // Latest available version is detected (15.0.2).
  eslint-config-next: 15.0.2
  react: 19.0.0-rc-02c0e824-20241028
  react-dom: 19.0.0-rc-02c0e824-20241028
  typescript: 5.6.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Which stage(s) are affected? (Select all that apply)

next build (local)

### Additional context

https://www.npmjs.com/package/critters

critters transferred ownership to beasties and marked the package as deprecated.
As the module import is in the nextjs logic, it is a bit of work to replace the module from Nextjs end user. Changing the module could also be a breaking change, not sure what is the best approach.

**Comment by Cajuteq** [comment]
Hi, I see this is one of the top discussions from last 12 month
It seems fairly straightforward (codewise) to replace critters by beasties as it is mainly an ownership change
The optimizeCss flag is under experimental so a breaking change here is kinda expected

So the question is what is holding back ? Would a PR tackling this be accepted ? What need to be discussed ?

**Comment by siimsams** [comment]
Hey!

There have been attempts before. But for some reason they were all closed or abandoned. 

Some examples are here:
https://github.com/vercel/next.js/pull/82437
https://github.com/austinderek/next.js/pull/197

[Here is my attempt to get this solved.](https://github.com/vercel/next.js/pull/88640) Reviews are welcome and I'll try to see it through till the end. 

Although it's just a build thing I just want to get rid of this warning in my repo. :D 
<img width="332" height="66" alt="Screenshot 2026-01-16 at 13 58 17" src="https://github.com/user-attachments/assets/9860d81d-1a9d-409f-a7bc-fec5602baa66" />

EDIT: 
Life got in the way but I'm back. I found a [bug in Beasties and created a fix](https://github.com/danielroe/beasties/pull/252). As soon as it's fixed I can finalize my PR and get it merged. 

EDIT2:
The fix in Critters was released. I updated the PR with the latest version. 🙌

EDIT3: 
It's outside of my hands. NextJS team must review and merge it.
